lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -207,6 +207,48 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     match_module_function(mod, hint, ctx) ++ match_module_type(mod, hint, ctx)
   end
 
+  defp match_struct_fields_or_local_or_var(hint, ctx) do
+    case expand_struct_fields(ctx) do
+      {:ok, struct, fields} ->
+        for field <- fields, do: {:map_field, field, nil}
+
+      _ ->
+        match_local_or_var(hint, ctx)
+    end
+  end
+
+  defp expand_struct_fields(ctx) do
+    with {:ok, quoted} <- Code.Fragment.container_cursor_to_quoted(ctx.hint),
+         {aliases, pairs} <- find_struct_fields(quoted),
+         alias = Enum.join(aliases, "."),
+         mod = expand_alias(alias, ctx),
+         true <- has_struct?(mod) do
+      fields =
+        pairs
+        |> Enum.reduce(Map.from_struct(mod.__struct__), fn {key, _}, map ->
+          Map.delete(map, key)
+        end)
+        |> Map.keys()
+        |> Enum.map(&Atom.to_string/1)
```

Comment:
Here we can keep fields as atoms, see `@type identifier_item` at the top.

Reply:
@jonatanklosko we're a bit inconsistent here, as you can see `:map_field` is returning a string as the field name.  I can make it consistent in this pr! https://github.com/livebook-dev/livebook/blob/main/lib/livebook/intellisense/identifier_matcher.ex#L241

Reply:
Oh wow, now I see we actually have strings for everything except function and module. It used to be strings, but then we needed the function name as atom, so I switched it to atom. Looks like for other types I updated the typespec and missed the actual items :0 Feel free to update all of those, but I can also do it if you prefer :)

Reply:
Oh I see that -- I think I'm going to get lost trying to do it all :)

Would you like me to back out https://github.com/livebook-dev/livebook/pull/793/commits/019c656b2c11010d24d1a61fefc448cf0426a512 so you can fix it all in one swoop?

Reply:
No worries, keep that commit and I will update the rest later :)